pkg/gitlab/gitlab_helper.go

```diff
@@ -219,3 +223,38 @@ func RefineGitHostingServiceError(response *http.Response, originErr error) erro
 func getDefaultBranch(client *GitlabClient, projectPath string) (string, error) {
 	return client.getDefaultBranch(projectPath)
 }
+
+// findUnmergedOnboardingMergeRequest finds out the unmerged merge request that is opened during the component onboarding
+// An onboarding merge request fulfills all the following criteria:
+// 1) opened based on the base branch which is determined by the Revision or is the default branch of component repository
+// 2) opened by a specific author
+// 3) opened from source branch appstudio-{component.Name}
+// If no onboarding merge request is found, nil is returned.
+func findUnmergedOnboardingMergeRequest(
+	glclient *GitlabClient, componentName, projectPath, baseBranch, authorName string) (*gitlab.MergeRequest, error) {
+	opts := &gitlab.ListProjectMergeRequestsOptions{
+		State:          gitlab.String("opened"),
+		AuthorUsername: gitlab.String(authorName),
+		SourceBranch:   gitlab.String(fmt.Sprintf("appstudio-%s", componentName)),
```

Comment:
should we keep the PaC init branch in user's repo or delete it as well?

Reply:
It should probably depend on users' preference. I think we can leave this flexibility to users at this moment.

Reply:
ok I thought about it, because if the branch is deleted, the PR is closed automatically (so 2 birds with one stone)